compiler-cli/src/new.rs

```diff
@@ -72,14 +72,22 @@ impl Creator {
         crate::fs::mkdir(&self.root)?;
         crate::fs::mkdir(&self.src)?;
         crate::fs::mkdir(&self.test)?;
-        crate::fs::mkdir(&self.github)?;
-        crate::fs::mkdir(&self.workflows)?;
-        crate::fs::git_init(&self.root)?;
+        if self.options.github_init {
+            crate::fs::mkdir(&self.github)?;
+            crate::fs::mkdir(&self.workflows)?;
+        }
+        if self.options.git_init {
+            crate::fs::git_init(&self.root)?;
+        }
 
         match self.options.template {
             Template::Lib => {
-                self.gitignore()?;
-                self.github_ci()?;
+                if self.options.git_init {
```

Comment:
I wasn't sure if this should be lumped in with `git init` functionality. What if the dev is using something other than git?